ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/JdbcLedgerDao.scala

```diff
@@ -110,11 +110,15 @@ private class JdbcLedgerDao(
 
   private val SQL_SELECT_LEDGER_END = SQL("select ledger_end from parameters")
 
-  override def lookupLedgerEnd(): Future[Offset] =
+  /**
+    * Defaults to Offset.begin if ledger_end is unset
+    */
+  override def lookupLedgerEnd(): Future[Offset] = {
     dbDispatcher.executeSql(metrics.daml.index.db.getLedgerEnd) { implicit conn =>
       SQL_SELECT_LEDGER_END
-        .as(offset("ledger_end").single)
+        .as(offset("ledger_end").?.map(_.getOrElse(Offset.begin)).single)
```

Comment:
I think that when this function is called, there should _always_ be a value here, so crashing would be appropriate. If we haven't initialized the ledger, the code should be calling `lookupInitialLedgerEnd` instead.

What happens if you revert this? What breaks?

Reply:
Yes, that's correct, @SamirTalwar-DA.

Reply:
The previous behaviour was:
1. If called prior to ledger initialisation - _crash_
2. If called post initialisation but pre first ledger end update - _default to Offset.begin_
3. If called post first ledger end update - _return most recently updated ledger_end_

This change means that the default provided in #2 above comes from the method as opposed to from the database column.

As many things could be happening asynchronously during startup I think it is safest to leave the default, or leave it someone more expert in the area to remove it ;-)